src/backend/access/transam/parallel.c

```diff
@@ -1680,6 +1680,9 @@ void GpDestroyParallelDSMEntry()
 
 	ParallelEntryTag tag;
 
+	if (!ParallelSession || !ParallelSession->segment)
+		return;
```

Comment:
seem lost logic：
```shell
	ParallelWorkerNumberOfSlice = -1;
	TotalParallelWorkerNumberOfSlice = 0;
```

Reply:
Fixed! Thanks.

Reply:
I just remember that there is no parallel on QD, so Gp_role seems the simplest way for this.

Reply:
If there is no parallel, the two variables won't be set, and don't need to be reset. Do we really need reset that?

Reply:
> If there is no parallel, the two variables won't be set, and don't need to be reset. Do we really need reset that?
We reset these two variables uniformly here ?

Reply:
I suggest reset,  that follow right logic.

Reply:
I agree with you.

Reply:
Reset them are useless instructions.

In fact, we can simply rely on `TotalParallelWorkerNumberOfSlice` to determine if calling this function is necessary.